### ISSUE TITLE
Remote HTTP references requiring authentication are reported as errors

### ISSUE THREAD

**Issue by denver-HJS** [stoplightio/spectral#1466]
**User story.**
As a Developer, I can provide my Stoplight credentials, so that I can lint my OAS spec that contains remote HTTP references guarded by Stoplight's (or any host's) access controls.

**Is your feature request related to a problem?**
I get linting errors when running spectral against a specification that's not located in Stoplight but contains remote references to Stoplight Next projects.

**Describe the solution you'd like**
I'd like to be able to provide my Stoplight application token to the CLI so that the remote references are resolved and incorporated into the linter instead of being reported as an error.

**Comment by P0lip** [comment]
Hey, it's possible to pass a custom $ref resolver, via `--resolver` CLI flag.

```
spectral lint --resolver my-resolver.js <document>
```

where `my-resolver.js` could look as follows

```js
'use strict';
const { Resolver } = require('@stoplight/json-ref-resolver');
const { resolveFile, createResolveHttp } = require('@stoplight/json-ref-readers');

const resolveHttp = createResolveHttp({
  headers: {
    // insert whatever header you need here
  }
});

module.exports = new Resolver({
  resolvers: {
    https: { resolve: resolveHttp },
    http: { resolve: resolveHttp },
    file: { resolve: resolveFile },
  },
});
```

**Comment by denver-HJS** [comment]
Awesome, thanks for the tip and examples @P0lip! I will try this out in my project.

**Comment by P0lip** [comment]
Awesome, let me know how it goes and if it worked out in the end!

**Comment by P0lip** [comment]
I'll close this one for now.
Do let me know if you succeeded and if any further assistance is needed - more than happy to help.
Thanks and have a good one!
